pkg/artifactory/repositories.go

```diff
@@ -853,15 +858,25 @@ func lookup(payload interface{}) map[string]interface{} {
 	for i := 0; i < t.NumField(); i++ {
 		field := t.Field(i)
 		thing := v.Field(i)
-		typeInspector := findInspector(thing.Kind())
-		for key, value := range typeInspector(field, thing) {
-			if _, ok := values[key]; !ok {
-				values[key] = value
+
+		shouldLookup := true
+		if thing.Kind() != reflect.Struct {
```

Comment:
The right thing to do is follow the functional paradigm and compose this as part of the logic. You basically have an `and` here

Reply:
Oh, I see, you shouldn't even be looking up the hcl on the field unless it's not a struct?

Reply:
Yea, the `struct` is from when we 'inherit' params from base struct. Without type checking, all attributes in the base params struct are skipped because the 'hcl' has the struct name.

Reply:
This check needs to happen for all/any predicates. Can't guarantee this logic will be implemented by all/any predicate.

Reply:
Actually it isn't just a simple `and`. When `thing.Kind()` is not a `struct`, we still want to perform the lookup.